### PR TITLE
Hotfix/build docs #33

### PR DESCRIPTION
**Pull request by LGro**
The docs build from the most recent release failed. To avoid a situation like that in the future, this PR adds a docs build to the testing stage to be executed already on PR and not just on merge. Also, it fixes the missing dependency.

### CODE REVIEW

**Review by sfalkner — APPROVED** [review]
Seems reasonable to test the docs as well. Thanks @LGro!

File: pyproject.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ pytest-rerunfailures = "^10.1"
 num2tex = "^0.8"
 mike = "^1.1.2"
 pydocstyle = "^6.1.1"
+ipython-genutils = "^0.2.0"
```
Comment (dynobo) [inline pyproject.toml:47]: Could we try bumping to `mkdocs-jupyter = "^0.21.0"` instead? `ipython-genutils` sound like being a sub-dep of this package which we pinned to a rather old version. Updating might solve this. 

If that doesn't work out in the first try, I'm also happy with this workaround.

File: .github/workflows/python.yaml [hunk 2]
```diff
@@ -36,6 +36,9 @@ jobs:
       - name: Run tests
         if: always()
         run: poetry run pytest
+      - name: Build docs
+        if: always()
+        run: poetry run mkdocs build
```
Comment (dynobo) [inline .github/workflows/python.yaml:41]: Very good idea!

**Review by dynobo — APPROVED** [review]
Thanks, Lukas!

**Pull request merged**